### DISCUSSION TITLE
Best practice for adding a check constraint in PostgresSQL?

### DISCUSSION THREAD

**Discussion by dmasters19** [prisma/prisma#16158] [Q&A] [answered]
If I use `npx prisma migrate dev` to generate the raw SQL migrations, but I want to add a custom check constraint as well, is it better to:

1. Simply add the constraint to the bottom of the auto-generated `migration.sql` file?
2. Create my own separate migration under the `migrations` directory with a new `migration.sql` file that only includes my new constraint?
3. Something else

**Comment by nurul3101** [comment, marked as answer]
Hello @dmasters19 👋 
You can create a migration file with `npx prisma migrate dev --create-only` command. This would create a migration file without applying it. Then you can add the check constraint at the bottom of the generated `migration.sql` file and then invoke `npx prisma migrate dev` command to apply all the pending migrations.

For reference here's a section of [Customizing Migrations](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#customizing-migrations)

Please let me know in case you run into any issues while following the customizing migrations reference.

**Reply by dmasters19** [reply]
Thank you @nurul3101 !

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
